src/stream.c

```diff
@@ -955,6 +955,31 @@ int stream_update(struct stream *s)
 }
 
 
+/**
+ * Removes the media encryption state from a stream.
+ *
+ * Only apply if SRTP module is used!
+ *
+ * The encryption consists of 1 encryption session state and N encryption
+ * media states.
+ *
+ * @param strm Stream to remove the media encryption state.
+ */
+void stream_remove_menc_media_state(struct stream *strm)
+{
+	if (!strm)
+		return;
+
+	if ((str_casecmp(strm->menc->id, "srtp") != 0) &&
+	    (str_casecmp(strm->menc->id, "srtp-mand") != 0) &&
+	    (str_casecmp(strm->menc->id, "srtp-mandf") != 0)) {
+		return;
+	}
```

Comment:
this goes against the modular design. The core should work with any module.
The core should not have any knowledge about module names or how they work.

Reply:
ok i see your point and remove the checks again.

as cleanup i would move it into core.h . So the function is only a private interface function. The testcases would have access to it since they include the core.h. And the function is only used in the corresponding testcase.

at least for now (until testcases are written) i can not say anything about the zrtp & dtls_srtp behaviour.